src/core/device.c

```diff
@@ -766,6 +766,28 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (u && UNIT_VTABLE(u)->active_state(u) == UNIT_ACTIVE) {
+                        r = manager_propagate_reload(m, u, JOB_REPLACE, NULL);
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to propagate reload: %m");
+                                return 0;
```

Comment:
This looks weird. If we indeed want to return 0, then why not drop this statement and go through the rest of the function? If there is a reason not to, shouldn't we return an error then?

Reply:
This function is called by the main loop through sd_event_add_io, so returning a negative error code would disable all events from udev. 

Propagating to the rest of the function could be done, but i'm not sure it's a good idea. At that point, the rest of the function is supposed to be a NOP since the unit is in an active state. 

Arguably, we could let pass through to make the rest of the code less confusing if someone changes it later.... 

Opinions welcomed

Reply:
If the rest of the function is supposed to be a NOP, then at least this return statement should move one if statement up.